app/models/report.rb

```diff
@@ -14,4 +23,37 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  def link_detect_and_save
+    extracted_report_ids = extract_report_ids(content)
+
+    extracted_report_ids.all? do |report_id|
+      mentioned_relationships = active_relationships.build(mentioned_id: report_id)
+      mentioned_relationships.save
+    end
+  end
+
+  def link_detect_and_update
+    extracted_report_ids = extract_report_ids(content)
+
+    old_report_ids = mentioning_reports.ids - extracted_report_ids
+    new_report_ids = extracted_report_ids - mentioning_reports.ids
+
+    old_report_ids.all? do |report_id|
+      delete_mentions = MentionedRelationship.find_by(mentioning_id: id, mentioned_id: report_id)
+      delete_mentions.destroy
+    end
+
+    new_report_ids.all? do |report_id|
+      mentioned_relationships = active_relationships.build(mentioned_id: report_id)
+      mentioned_relationships.save
+    end
+  end
+
+  private
+
+  def extract_report_ids(text)
+    urls = text.scan(%r{http://localhost:3000/reports/\d+\b})
```

Comment:
```suggestion
    urls = text.scan(%r{http://localhost:3000/reports/(\d+)\b})
```

のようにしてidの部分だけをキャプチャするともうちょっとスッキリ書けるかも

Reply:
なるほど、そんなことができるのですね！変更しました！